src/datasources/balances-api/safe-balances-api.service.ts

```diff
@@ -108,10 +114,54 @@ export class SafeBalancesApi implements IBalancesApi {
     await this.cacheService.deleteByKey(key);
   }
 
-  /**
-   * No fiat prices are available from this provider.
-   */
-  getFiatCodes(): string[] {
-    return [];
+  async getFiatCodes(): Promise<string[]> {
+    return this.coingeckoApi.getFiatCodes();
+  }
+
+  private async _mapBalances(
+    balances: Balance[],
+    fiatCode: string,
+  ): Promise<Balance[]> {
+    const tokenAddresses = balances
+      .map((balance) => balance.tokenAddress)
+      .filter((address): address is string => address !== null);
+
+    const prices = await this.coingeckoApi.getTokenPrices({
+      chainId: this.chainId,
+      fiatCode,
+      tokenAddresses,
+    });
+
+    return await Promise.all(
+      balances.map(async (balance) => {
+        const tokenAddress = balance.tokenAddress?.toLowerCase() ?? null;
+        let price: number | null;
+        if (tokenAddress === null) {
+          price = await this.coingeckoApi.getNativeCoinPrice({
+            chainId: this.chainId,
+            fiatCode,
+          });
+        } else {
+          const found = prices.find((assetPrice) => assetPrice[tokenAddress]);
+          price = found?.[tokenAddress]?.[fiatCode.toLowerCase()] ?? null;
+        }
+        const fiatBalance = await this._getFiatBalance(price, balance);
+        return {
+          ...balance,
+          fiatBalance: fiatBalance ? getNumberString(fiatBalance) : null,
+          fiatConversion: price ? getNumberString(price) : null,
+        };
+      }),
+    );
+  }
+
+  private _getFiatBalance(
+    price: number | null,
+    balance: Balance,
+  ): number | null {
+    return price !== null
+      ? (price * Number(balance.balance)) /
+          10 ** (balance.token?.decimals ?? 18)
```

Comment:
I'd suggest extracting some of these numbers as constants within the function.

Reply:
Do you mean extracting the hardcoded values? I've extracted `18` to a static class variable (`DEFAULT_DECIMALS`) in [a62b130](https://github.com/safe-global/safe-client-gateway/pull/1175/commits/a62b1305da4626e6b4362831001b84289acfed44)

Or do you mean extracting intermediate results to variables within the function?

Reply:
The `18` is what I mean. Thanks!